lib/middleware.js

```diff
@@ -347,6 +348,42 @@ class Kevin {
                 return;
             }
 
+            if (req.method === "POST" && reqPath === `${this.kevinApiPrefix}/restart-compiler`) {
+                // This endpoint restarts compilers. It accepts two query params:
+                // `compiler` (required), which is the name of the compiler, and
+                // `hard` (optional) which, if true, fully shuts down and restarts the compiler,
+                //          rather than just invalidating it and forcing a partial recompilation.
+                if (!req.query.compiler || !manager.isCompilerActive(req.query.compiler)) {
+                    return res.status(400).send("Kevin couldn't find a compiler with that name");
```

Comment:
Is it worth including the provided compiler name and also the list of active compilers?

Reply:
I'll add the name, but I'd rather not list existing compilers. We have two other endpoints that do that programmatically already (`/__kevin/build-status` and `/__kevin/compiler-info`) and I'd like to make sure people use that rather than trying to parse the error or something.